cwarhm/data_specific_processing/era5.py

```diff
@@ -0,0 +1,227 @@
+                        dest[name_summa].setncattr('missing_value' ,-999)
+                        dest[name_summa].setncatts(loop_attr_copy_values)
+
+                        # 3c. Copy the data SECOND
+                        dest[name_summa][:] = loop_val
+
+                # === Transfer the pressure level variables next, using the same procedure as above
+                for name, variable in src1.variables.items():
+                    if name in variables_pres_convert:
+
+                        # 0. Reset the dictionary that we keep attribute values in
+                        loop_attr_source_values = {name: 'n/a' for name in attr_names_expected}
+
+                        # 1a. Get the values of this variable from the source (this automatically applies scaling and offset)
+                        loop_val = variable[:]
+
+                        # 1b. Get the attributes for this variable from source
+                        for attrname in variable.ncattrs():
+                            loop_attr_source_values[attrname] = variable.getncattr(attrname)
+
+                        # 2a. Find what this ERA5 variable should be called in SUMMA
+                        if name == 't':
+                            name_summa = 'airtemp'
+                        elif name == 'q':
+                            name_summa = 'spechum'
+                        elif name == 'u':
+                            name_summa = 'n/a/' # we shouldn't reach this part of the code, because 'u' is not specified in 'variables_pres_convert'
+                        elif name == 'v':
+                            name_summa = 'n/a' # as with 'u', because both are needed to calculate total wind speed first
+                        else:
+                            name_summa = 'n/a/' # no name so we don't start overwriting data if a new name is not defined for some reason
+
+                        # 2b. Create the .nc variable with the proper SUMMA name
+                        # Inputs: variable name as needed by SUMMA; data type: 'float'; dimensions; no need for fill value, because thevariable gets populated in this same script
+                        dest.createVariable(name_summa, 'f4', ('time' ,'latitude' ,'longitude'), fill_value = False)
+
+                        # 3a. Select the attributes we want to copy for this variable, based on the dictionary defined before the loop starts
+                        loop_attr_copy_values = {use_this: loop_attr_source_values[use_this] for use_this in loop_attr_copy_these}
+
+                        # 3b. Copy the attributes FIRST, so we don't run into any scaling/offset issues
+                        dest[name_summa].setncattr('missing_value' ,-999)
+                        dest[name_summa].setncatts(loop_attr_copy_values)
+
+                        # 3c. Copy the data SECOND
+                        dest[name_summa][:] = loop_val
+
+                # === Calculate combined wind speed and store
```

Comment:
calculating combined wind speed is ideally also a separate function, not in scope for now.

Reply:
Yes, now the main task is conversion.